### ISSUE TITLE
[Feature] Improve the automated investment pattern.

### ISSUE THREAD

**Issue by michael-spengler** [michael-spengler/decentralized-finance#1]
**Prelude**
The following proposals shall be understood considering the "net cash on average" having the reinvestment rounds or redemption rounds already in the equation.

**Optimize the Degree of the Leverage Effect**
It seems to be a good idea to automatically and slowly decrease the leverage effect along times of rising asset prices in order to protect the investment during times of falling asset prices.
According to that it is probably reasonable to automatically and slowly increase the leverage effect along times of falling asset prices in order to increase the probability of generating a strong outcome.

**In other words** 
The health factor of an investment should increase during rising asset prices 
It is probably also reasonable that it decreases during falling asset prices 

**Implementation**
Shall be done in the following class: https://github.com/michael-spengler/decentralized-finance/blob/main/src/thug-life-investments/thug-life.service.ts

**Comment by michael-spengler** [comment]
**Just an additional optional pattern:**
Combine investing with artificial neural net based trading. The neural net shall leverage the following kinds of training data:
1. wale wallet transactions
2. sentiment analysis
3. classical trader stuff like deriving future price development guesses from past prices and price correlations (https://www.investopedia.com/terms/a/atr.asp...)

Explore and leverage the following technological assets:
1. https://www.npmjs.com/package/brain.js
2. https://www.tensorflow.org/js
3. https://deno.land/x/neural_nets@0.0.3
4. https://www.npmjs.com/package/node-nlp (for the sentiment analysis)
